1_fetch/src/my_happy_script.R

```diff
@@ -127,43 +124,33 @@ render_data <- list(pgdl_980mean = filter(eval_data, model_type == 'pgdl', exper
                     pb_100mean = filter(eval_data, model_type == 'pb', exper_id == "similar_100") %>% pull(rmse) %>% mean %>% round(2),
                     pgdl_2mean = filter(eval_data, model_type == 'pgdl', exper_id == "similar_2") %>% pull(rmse) %>% mean %>% round(2),
                     pb_2mean = filter(eval_data, model_type == 'pb', exper_id == "similar_2") %>% pull(rmse) %>% mean %>% round(2))
-return(render_data)
-     }
-
-whiskerRender <- function(dataRender){
+     
 template_1 <- 'resulted in mean RMSEs (means calculated as average of RMSEs from the five dataset iterations) of {{pgdl_980mean}}, {{dl_980mean}}, and {{pb_980mean}}°C for the PGDL, DL, and PB models, respectively.
   The relative performance of DL vs PB depended on the amount of training data. The accuracy of Lake Mendota temperature predictions from the DL was better than PB when trained on 500 profiles 
   ({{dl_500mean}} and {{pb_500mean}}°C, respectively) or more, but worse than PB when training was reduced to 100 profiles ({{dl_100mean}} and {{pb_100mean}}°C respectively) or fewer.
   The PGDL prediction accuracy was more robust compared to PB when only two profiles were provided for training ({{pgdl_2mean}} and {{pb_2mean}}°C, respectively). '
 
-# dataRender() is supposed to be the input here
-whisker.render(template_1 %>% str_remove_all('\n') %>% str_replace_all('  ', ' '), dataRender ) %>% cat(file = file.path(project_output_dir, 'model_diagnostic_text.txt'))
+# since this is inside the dataRender function, this should be able to use the local variable render_data
+whisker.render(template_1 %>% str_remove_all('\n') %>% str_replace_all('  ', ' '), render_data) %>% cat(file = file.path(project_output_dir, 'model_diagnostic_text.txt'))
 }
 
-
+########
+# DIRECTIONS :
 # run this code block after adding each function to the namespaces:
 
-########
-getCSV()
-#create object for evaluated data:
-evalData <- eval_data(getCSV())
-thePlot <- createPlot(evalData)             # returns "null device 1, null device is always device 1, I think it aludes to the dev.off for the .png. "
-thePlot                                     # returns "object not found"
+# obtain data from URL:
+getCSV(out_folder)
 
-# not sure which of these formats to do, the top one throws an error, this error can be resolved 
-# with passing the inner as a function rather than the function's name. Passing the function however doesn't output the file.
+#create object for evaluated data:
+evalData <- eval_data(getCSV(out_folder))
```

Comment:
Lines 142 and 145 will download the data twice. You could do the following to avoid that:

```
file_out <- getCSV(out_folder)
evalData <- eval_data(file_out)
```